libs/vkd3d/resource.c

```diff
@@ -532,13 +533,24 @@ HRESULT vkd3d_get_image_allocation_info(struct d3d12_device *device,
     }
 
     /* XXX: We have to create an image to get its memory requirements. */
-    if (SUCCEEDED(hr = vkd3d_create_image(device, &heap_properties, 0, desc, NULL, &vk_image)))
-    {
-        VK_CALL(vkGetImageMemoryRequirements(device->vk_device, vk_image, &requirements));
-        VK_CALL(vkDestroyImage(device->vk_device, vk_image, NULL));
+    if (FAILED(hr = vkd3d_create_image(device, &heap_properties, 0, desc, NULL, &vk_image)))
+        return hr;
+
+    VK_CALL(vkGetImageMemoryRequirements(device->vk_device, vk_image, &requirements));
+    VK_CALL(vkDestroyImage(device->vk_device, vk_image, NULL));
 
-        allocation_info->SizeInBytes = requirements.size;
-        allocation_info->Alignment = requirements.alignment;
+    allocation_info->SizeInBytes = requirements.size;
+    allocation_info->Alignment = requirements.alignment;
+
+    /* Do not report alignments greater than DEFAULT_RESOURCE_PLACEMENT_ALIGNMENT
+     * since that might confuse apps. Instead, pad the allocation so that we can
+     * align the image ourselves. */
+    max_alignment = desc->Alignment ? desc->Alignment : D3D12_DEFAULT_RESOURCE_PLACEMENT_ALIGNMENT;
```

Comment:
Shouldn't this be a max()?

Reply:
nvm. I think the use of `max_alignment` is confusing wording here, `target_alignment` might make more sense.